zipline/data/_minute_bar_internal.pyx

```diff
@@ -37,10 +37,50 @@ def minute_value(ndarray[long_t, ndim=1] market_opens,
     return market_opens[q] + r
 
 def find_position_of_minute(ndarray[long_t, ndim=1] market_opens,
+                            ndarray[long_t, ndim=1] market_closes,
```

Comment:
this whole diff confused me until I realized you had simply renamed the original `find_position_of_minute ` method to `find_position_of_minute_half_day_unadjusted` and added some logic in there to be resilient against day barriers.  I was confused why you had been changing the `IntervalTree` logic elsewhere.

I think it would be easier to parse if we kept `find_position_of_minute` with the same name (and add the small piece of new logic in there), and create a new method called `find_position_of_minute_with_halfdays` that is the totally new method.  I can do this later today if needed.